ChattoApp/ChattoApp/Source/Chat View Controllers/DemoChatViewController.swift

```diff
@@ -239,8 +257,190 @@ class DemoChatViewController: UIViewController {
+        ]
+    }
+
+    private static func makeNewImagePresenterBuilder() -> ChatItemPresenterBuilderProtocol {
+
+        var factory = FactoryAggregate<DemoPhotoMessageModel>()
+
+        let image = factory.register(
+            viewFactory: AsyncImageViewFactory(),
+            viewModelFactory: AsyncImageViewModelFactory(),
+            layoutProviderFactory: AsyncImageLayoutProviderFactory()
+        )
+
+        let bubble = factory.register(
+            viewFactory: MessageBubbleViewFactory(),
+            viewModelFactory: MessageBubbleViewModelFactory(),
+            layoutProviderFactory: MessageBubbleLayoutProviderFactory(configuration: .init(percentageToOccupy: 0.4))
+        )
+
+        var binder = Binder()
+
+        binder.registerBinding(for: bubble)
+
+        binder.registerBlockBinding(for: image) { view, viewModel in
+            view.viewModel = viewModel
+        }
+
+        var assembler = ViewAssembler(root: bubble)
+        assembler.register(child: image, parent: bubble)
+
+        var layoutAssembler = LayoutAssembler(rootKey: bubble)
+        layoutAssembler.register(child: image, for: bubble)
+
+        // TODO: Remove #746
+        layoutAssembler.populateSizeProviders(for: bubble)
+        layoutAssembler.populateSizeProviders(for: image)
+
+        return ChatItemPresenterBuilder(
+            binder: binder,
+            assembler: assembler,
+            layoutAssembler: layoutAssembler,
+            factory: factory
+        )
+    }
+
+    private static func makeNewCompoundExamplePresenterBuilder() -> ChatItemPresenterBuilderProtocol {
+        var factory = FactoryAggregate<DemoCompoundMessageModel>()
+
+        let bubble = factory.register(
+            viewFactory: MessageBubbleViewFactory(),
+            viewModelFactory: MessageBubbleViewModelFactory(),
+            layoutProviderFactory: MessageBubbleLayoutProviderFactory(configuration: .init(percentageToOccupy: 0.8))
+        )
+
+        let compound = factory.register(
+            viewFactory: CompoundViewFactory(),
+            viewModelFactory: CompoundViewModelFactory(),
+            layoutProviderFactory: CompoundLayoutProviderFactory()
+        )
+
+        let first = factory.register(
+            viewFactory: DummyContentViewFactory(),
+            viewModelFactory: StaticDummyViewModelFactory(text: "first"),
+            layoutProviderFactory: DummyLayoutProviderFactory()
+        )
+
+        let second = factory.register(
+            viewFactory: DummyContentViewFactory(),
+            viewModelFactory: StaticDummyViewModelFactory(text: "second"),
+            layoutProviderFactory: DummyLayoutProviderFactory()
+        )
+
+        var binder = Binder()
+
+        binder.registerBinding(for: bubble)
+        binder.registerNoopBinding(for: compound)
+        binder.registerBinding(for: first)
+        binder.registerBinding(for: second)
+
+        var assembler = ViewAssembler(root: bubble)
```

Comment:
Will the view assembler make the same registrations as the layout assembler. Can we avoid this duplication somehow?

Reply:
Going to tackle that one in #749